src/github.com/getlantern/lantern-android/client/client.go

```diff
@@ -107,13 +106,16 @@ func (client *MobileClient) recordAnalytics() {
 	// is a little unorthodox by Lantern standards because it doesn't
 	// pin the certificate of the cloud.yaml root CA, instead relying
 	// on the go defaults.
-	httpClient, er := util.HTTPClient("", client.Client.Addr)
-	if er != nil {
-		log.Fatalf("Could not create HTTP client %v", er)
+	httpClient, err := util.HTTPClient("", client.Client.Addr)
+	if err != nil {
+		log.Fatalf("Could not create HTTP client %v", err)
 	} else {
+		if err := waitforserver.WaitForServer("tcp", client.Client.Addr, 3*time.Second); err != nil {
+			log.Print(err)
+			return
+		}
```

Comment:
This one shouldn't actually be necessary because we're doing it in onListening, so we know the server's up at this point.